client/src/features/Event/New/hooks/useBasicEventForm.ts

```diff
@@ -0,0 +1,68 @@
+import { useState, useMemo, useEffect } from 'react';
+
+import { BasicEventFormFields } from '../../types/Event';
+import { FIELD_CONFIG } from '../constants/formFieldConfig';
+import { validateEventForm } from '../utils/validateEventForm';
+
+export const useBasicEventForm = () => {
+  const [basicForm, setBasicForm] = useState<BasicEventFormFields>({
+    title: '',
+    eventStart: '',
+    eventEnd: '',
+    registrationEnd: '',
+    place: '',
+    description: '',
+    maxCapacity: 0,
+  });
+  const [touchedMap, setTouchedMap] = useState<
+    Partial<Record<keyof BasicEventFormFields, boolean>>
+  >({});
+  const [errors, setErrors] = useState<Record<string, string>>({});
+
+  useEffect(() => {
+    const validation = validateEventForm(basicForm);
+    setErrors(validation);
+  }, [basicForm]);
+
+  const setTouched = (key: keyof BasicEventFormFields) => {
+    setTouchedMap((prev) => ({ ...prev, [key]: true }));
+  };
+
+  const setValue = <K extends keyof BasicEventFormFields>(
+    key: K,
+    value: BasicEventFormFields[K]
+  ) => {
+    setBasicForm((prev) => ({ ...prev, [key]: value }));
+
+    const config = FIELD_CONFIG[key];
+    if (config?.required && typeof value === 'string' && value.trim() === '') {
+      setTouchedMap((prev) => ({ ...prev, [key]: true }));
+    }
+  };
+
+  const validate = () => {
+    const validation = validateEventForm(basicForm);
+    setErrors(validation);
+    return Object.keys(validation).length === 0;
+  };
+
+  const validateField = (key: keyof BasicEventFormFields, value: string | number) => {
+    const updated = { ...basicForm, [key]: value };
+    const validation = validateEventForm(updated);
+
+    setErrors((prev) => ({ ...prev, [key]: validation[key] || '' }));
+  };
+
+  const isValid = useMemo(() => Object.keys(errors).every((k) => !errors[k]), [errors]);
```

Comment:
p3.
집약된 표현보다는 명확하게 풀어서 작성해주면 이해하는 데 더 도움이 될 것 같아요.
현재는 단순히 배열을 반환하고 있어서 useMemo가 꼭 필요한 상황은 아닌 것 같은데,
에리얼이 useMemo를 사용하신 이유가 있으신가요?? 혹은 useMemo를 사용하는 기준이 있는지 궁금해요!

```suggestion
  const isValid = useMemo(() => Object.keys(errors).every((err) => !errors[er]), [errors]);
```

Reply:
특정 값들이 바뀔 때만 계산되어야 하는 게 명확하다면 종종 useMemo를 사용하는 것 같긴 한데요,
실제로는 성능 개선 효과가 크지 않은 경우도 많아서, 언제 도입해야 하는지 기준이 애매하다고 느껴질 때가 많은 것 같아요.
세라는 혹시 기준이 있나용??